src/curvemap.cpp

```diff
@@ -0,0 +1,678 @@
+	SI fn )		// field number of polynomial subrecord in ah
+{
+	return normalize( ah, fn, "rated conditions of ahccDsTWbEn", "ahccDsTDbCnd", ah->ahcc.dsTWbEn, ah->ahcc.dsTDbCnd);
+
+}		// PYBIQUAD::normalizeCoil
+//=============================================================================================================================
+RC PYBIQUAD::normalize( 		// if biquadratic poly coefficients are inconsistent, normalize and/or issue message.
+
+	record *r, 			// non-sub record containing the poly subrecord -- needed re error message
+	SI fn,			// field number of polynomial subrecord in r
+	const char* descX, const char* descY,	// descriptive texts, for use in "for %s=%g and %s=%g"
+	DBL x, DBL y,   		// argument value for which poly should be 1.0
+	BOO noWarn /*=FALSE*/ )	// true to suppress normalization warning (eg if defaulted & coeff known non-normal)
+// returns non-RCOK to stop run (error message already issued)
+{
+	float v = val(x,y);				// get polynomial value for inputs x,y
+	fn += PYBIQUAD_K;				/* adjust to field number of k[], what user input & what is in CULT tables,
+						   for 4 uses in error calls below. */
+#ifdef NORMALIZE
+	if (v != 0)  				// if v is 0, normalizing would divide by 0.  Fall thru to error message
+	{
+		for (SI i = 0; i < 6; i++)  		// six coefficients
+			k[i] /= v;				// normalize coefficients
+#if 1						// omit if normalize warning not desired
+		if ( !noWarn				// if warning not suppressed
+				&&  fabs(v - 1.) > .02 )   		// if much wrong, tell user, at least for now 7-14-92
+			return r->oInfo( 			// returns RCOK 7-92
+						  MH_S0689,	/* "Inconsistent '%s' coefficients: value is %g, not 1.0,\n"
+					   "    for %s=%g and %s=%g.\n"
+					   "    Coefficients will be normalized and execution will continue." */
+						  r->mbrIdTx( fn),	// get id text for record/field. slow: do not do til error detected.
+						  v, descX, x, descY, y );
+#endif
+	}
+	else					// zero value is error: cannot normalize (would /0)
+#else
+	if (v==0. ||  fabs(v - 1.) > .001)
+#endif
+		return r->ooer( fn,			// returns RCBAD
+					 MH_S0690,		/* "Inconsistent '%s' coefficients: value is %g, not 1.0,\n"
+					   "    for %s=%g and %s=%g\n" */
+					 r->mbrIdTx( fn),   	// get id text for record/field. slow: do not do til error detected.
+					 v, descX, x, descY, y );
+	return RCOK;
+}			// PYBIQUAD::normalize
+//=============================================================================
+
+///////////////////////////////////////////////////////////////////////////////
+// class PMACCESS: supports runtime access to perf map
+//                 retains pointer to Btwxt interpolator
+// class PERFORMANCEMAP: input representation of performance map
+//      class PMGRIDAXIS:  perfmap grid values (independent vars, e.g. DBT, speed, )
+//      class PMLOOKUPDATA:  perfmap performance data (e.g. capacity, power, )
+///////////////////////////////////////////////////////////////////////////////
+PMACCESS::PMACCESS()
```

Comment:
This class should be renamed so it's clear it is specific to the RSYS use-case.

Reply:
Maybe.  Wait for typed performance map redesign and see what falls out.